kmip/services/kmip_client.py

```diff
@@ -217,6 +217,7 @@ def open(self):
             suppress_ragged_eofs=self.suppress_ragged_eofs)
         self.protocol = KMIPProtocol(self.socket)
 
+        self.socket.settimeout(self.timeout)
```

Comment:
Regarding the unstable behaviour, iam not sure thats the case, the same socket can be set with a different timeout in a different context. KMIP as per protocol is TTLV and not a transport variant and so doesnt make sense for it to specify the timeout as per of specification. In anyways, i believe, that should not a big issue to deal with now.